### PR TITLE
quiche: enable more quic tests

### PR DESCRIPTION
**Pull request by danzh2010**
Commit Message: remove or comment on EXCLUDE_DOWNSTREAM_HTTP3 in various integration tests

Risk Level: low
Testing: existing integration tests

Part of #12930

### CODE REVIEW

File: test/integration/http_integration.cc [hunk 1]
```diff
@@ -254,11 +254,24 @@ IntegrationCodecClientPtr HttpIntegrationTest::makeRawHttpConnection(
...
-        http2_options.value().override_stream_error_on_invalid_http_message().value());
-  } else if (http2_options.value().stream_error_on_invalid_http_messaging()) {
-    http3_options.mutable_override_stream_error_on_invalid_http_message()->set_value(true);
+  } else {
+#ifdef ENVOY_ENABLE_QUIC
+    if (http2_options->has_initial_stream_window_size() &&
+        http2_options->initial_stream_window_size().value() < quic::kStreamReceiveWindowLimit) {
+      // Set http3 stream flow control window only if the configured http2 stream flow control
+      // window is smaller than the upper limit of flow control window supported by QUICHE which is
+      // also the default for http3 stream.
```
Comment (RyanTheOptimist) [inline test/integration/http_integration.cc:263]: nit: "http3 steam" => "http3 streams", I think?

**Review by RyanTheOptimist — COMMENTED** [review]
Woo hoo! Sweet!

File: test/integration/http_integration.cc (hunk 1 above)
Comment (danzh2010) [reply test/integration/http_integration.cc]: fixed

File: test/integration/multiplexed_integration_test.cc [hunk 2]
```diff
@@ -147,7 +147,7 @@ TEST_P(Http2IntegrationTest, CodecStreamIdleTimeout) {
 }
 
 TEST_P(Http2IntegrationTest, Http2DownstreamKeepalive) {
-  EXCLUDE_DOWNSTREAM_HTTP3;
+  EXCLUDE_DOWNSTREAM_HTTP3; // Need to support keepalive.
```
Comment (alyssawilk) [inline test/integration/multiplexed_integration_test.cc:150]: Do we have a tracking issue for this, and testing mid-stream codec errors?

File: test/integration/http_integration.cc [hunk 3]
```diff
@@ -254,11 +254,24 @@ IntegrationCodecClientPtr HttpIntegrationTest::makeRawHttpConnection(
         envoy::config::core::v3::Http2ProtocolOptions());
     http2_options.value().set_allow_connect(true);
     http2_options.value().set_allow_metadata(true);
-  } else if (http2_options.value().has_override_stream_error_on_invalid_http_message()) {
-    http3_options.mutable_override_stream_error_on_invalid_http_message()->set_value(
-        http2_options.value().override_stream_error_on_invalid_http_message().value());
-  } else if (http2_options.value().stream_error_on_invalid_http_messaging()) {
-    http3_options.mutable_override_stream_error_on_invalid_http_message()->set_value(true);
+#ifdef ENVOY_ENABLE_QUIC
+  } else {
```
Comment (alyssawilk) [inline test/integration/http_integration.cc:258]: I'm fine doing this inline but I wonder if it should be a static config utility function.  WDYT?

**Review by alyssawilk — APPROVED** [review]
LGTM, though I'm going to hold merging until we have a tracking issue

File: test/integration/multiplexed_integration_test.cc (hunk 2 above)
Comment (alyssawilk) [reply test/integration/multiplexed_integration_test.cc]: junk data in the middle of an active stream.

**Review by alyssawilk — APPROVED** [review]

**Comment by alyssawilk** [comment]
coverage failure was an unrelated flake - kicking off another run

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**